dotCMS/src/main/java/com/dotmarketing/portlets/contentlet/model/Contentlet.java

```diff
@@ -222,15 +223,18 @@ public String getCategoryId() {
     	return getInode();
     }
 
-    /**
-     * Create a contentlet based on a map (makes a copy of it)
-     * @param map
-     */
-    public Contentlet(final Map<String, Object> map) {
-		this.map = new ContentletHashMap();
-    	this.map.putAll(map);
-		this.indexPolicy = IndexPolicy.DEFER;
-    }
+  /**
+   * Create a contentlet based on a map (makes a copy of it)
+   * 
+   * @param map
+   */
+  public Contentlet(final Map<String, Object> mapIn) {
+
+    mapIn.values().removeIf(Objects::isNull);
```

Comment:
removing any null values from the incoming map - they cause an exception to be thrown.

Reply:
Very cool